python/pyarrow/tests/conftest.py

```diff
@@ -219,3 +230,66 @@ def tempdir(tmpdir):
 @pytest.fixture(scope='session')
 def datadir():
     return pathlib.Path(__file__).parent / 'data'
+
+
+try:
+    from tempfile import TemporaryDirectory
+except ImportError:
+    import shutil
+
+    class TemporaryDirectory(object):
+        """Temporary directory implementation for python 2"""
```

Comment:
Can also avoid bothering with this and make `s3fs` Python 3-only if that's easier.

Reply:
It works now, and only affects the test suite, so we can keep it.